### ISSUE TITLE
 `-march=native` compile flag cause program cannot run on another device

### ISSUE THREAD

**Issue by WorstCodeWay** [PointCloudLibrary/pcl#5883]
**Describe the bug**

It's a platform related bug. I build my program in a device(x86) and cannot run it in another device(x86 with different instruction sets). And finally, I find it's caused by `-march=native` added by PCL which can verified by change it into `-march=x86-64`  optimaizing with generic sets. But this modification cannot be done successfully in cmake (cause my lacking of  cmake knowledge)

**Context**

Change complie flag from `-march=native`  to `-march=x86-64`  in my program cmake file.

**Expected behavior**
Program built with generic instructions

**Current Behavior**

Optimizing with `-march=native` 

**To Reproduce**

Provide a link to a live example, or an unambiguous set of steps to reproduce this bug. A reproducible example helps to provide faster answers. If you load data e.g. from a PCD or PLY file, please provide the file.

**Screenshots/Code snippets**

In order to help explain your problem, please consider adding
* screenshots of the GUI issues
* code snippets: [syntax for code](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code) with correct language highlights

**Your Environment (please complete the following information):**

 - OS: Ubuntu 20.04
 - Compiler: g++ 9.4.0
 - PCL Version: 1.10

**Possible Solution**

Not obligatory, but suggest a fix/reason for the bug. Feel free to create a PR if you feel comfortable.

**Additional context**

**Comment by larshg** [comment]
You can disable march native by setting PCL_ENABLE_MARCHNATIVE to False. Ie when you call cmake add -DPCL_ENABLE_MARCHNATIVE=False.

Then SSE and AVX will still be enabled, which might also needs to be disabled. Depending on the target CPU.

**Comment by WorstCodeWay** [comment]
> You can disable march native by setting PCL_ENABLE_MARCHNATIVE to False. Ie when you call cmake add -DPCL_ENABLE_MARCHNATIVE=False.
> 
> Then SSE and AVX will still be enabled, which might also needs to be disabled. Depending on the target CPU.

Thanks for quick reply, and I'm sorry for new to cmake. When should I add it？Building pcl or  my own program? I tried when building my own program like:
```
cmake -DPCL_ENABLE_MARCHNATIVE=False -DCMAKE_BUILD_TYPE=Release ..
```
but the final compile is still `-march=native`

**Comment by larshg** [comment]
No, you need to do it when you build PCL.

**Comment by WorstCodeWay** [comment]
@larshg I did it just now with following cmake call, but result from cmake implies it still using `native`, am I right?
```cmake
cmake -DPCL_ENABLE_MARCHNATIVE=False -DCMAKE_BUILD_TYPE=Release ..
```

Result contains
```
...
-- Using CPU native flags for SSE optimization: -march=native
...
```

or I can just ignore that? It indeed disable the native flag?

**Comment by mvieth** [comment]
@WorstCodeWay Did you do a completely new build, meaning did you remove all files created by CMake beforehand? Otherwise the `march=native` might be stored in some cache variable.

**Comment by WorstCodeWay** [comment]
> @WorstCodeWay Did you do a completely new build, meaning did you remove all files created by CMake beforehand? Otherwise the `march=native` might be stored in some cache variable.

@mvieth Yes, I recompiled from stretch by removing build directory and recreate it. However, my problem solved by adding both `-DPCL_ENABLE_MARCHNATIVE=False` and `-DPCL_ENABLE_MARCHNATIVE=False -DPCL_ENABLE_SSE=False`, I don't know if it is as right?

**Comment by mvieth** [comment]
@WorstCodeWay Which PCL version are you compiling? `PCL_ENABLE_MARCHNATIVE` is only available since PCL version 1.12.1. If you use PCL 1.10 as you wrote above, then `-DPCL_ENABLE_SSE=False` is indeed the way to go.
